lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -997,76 +1084,55 @@ public function orderBy(string $sort, ?string $order = null) : self
      */
     public function addOrderBy(string $sort, ?string $order = null) : self
     {
-        return $this->add('orderBy', $sort . ' ' . (! $order ? 'ASC' : $order), true);
-    }
+        $orderBy = $sort;
 
-    /**
-     * Gets a query part by its name.
-     *
-     * @return mixed
-     */
-    public function getQueryPart(string $queryPartName)
-    {
-        return $this->sqlParts[$queryPartName];
-    }
+        if ($order !== null) {
+            $orderBy .= ' ' . $order;
+        }
 
-    /**
-     * Gets all query parts.
-     *
-     * @return array<string, mixed>
-     */
-    public function getQueryParts() : array
-    {
-        return $this->sqlParts;
+        $this->orderBy[] = $orderBy;
+
+        $this->state = self::STATE_DIRTY;
+
+        return $this;
     }
 
     /**
-     * Resets SQL parts.
-     *
-     * @param array<int, string>|null $queryPartNames
-     *
-     * @return $this This QueryBuilder instance.
+     * @throws QueryException
      */
-    public function resetQueryParts(?array $queryPartNames = null) : self
+    private function getSQLForSelect() : string
     {
-        if ($queryPartNames === null) {
-            $queryPartNames = array_keys($this->sqlParts);
+        if (! $this->select) {
```

Comment:
Please use `(if count($this->select) < 1)` instead. If at some moment `$this->select` is replaced with an `interable&coutable` object, this logic will break.

Reply:
As long as they're arrays, I'm perfectly fine with implicit conversion to boolean, but you may have a point. I had Scrutinizer's suggestion to use `empty()` though: it lets undefined variables pass through.

Anyway, done.

Reply:
`empty()` is a ticking bomb. Should be avoided, especially in new code.